hazelcast-client/src/test/java/com/hazelcast/client/executor/durable/ClientDurableRetrieveResultTest.java

```diff
@@ -0,0 +1,160 @@
+
+        client.shutdown();
+        client = hazelcastFactory.newHazelcastClient();
+
+        executorService = client.getDurableExecutorService(name);
+        Future<Boolean> future = executorService.retrieveAndDisposeResult(taskId);
+        assertTrue(future.get());
+
+        Future<Object> f = executorService.retrieveResult(taskId);
+        assertNull(f.get());
+    }
+
+    @Test
+    public void testRetrieveAndDispose_WhenOwnerMemberDown() throws ExecutionException, InterruptedException {
+        String name = randomString();
+        String key = generateKeyOwnedBy(instance2);
+
+        DurableExecutorService executorService = client.getDurableExecutorService(name);
+        SleepingTask task = new SleepingTask(4);
+        long taskId = executorService.submitToKeyOwner(task, key).getTaskId();
+        instance2.shutdown();
+
+        Future<Boolean> future = executorService.retrieveAndDisposeResult(taskId);
+        assertTrue(future.get());
+
+        Future<Boolean> f = executorService.retrieveResult(taskId);
+        assertNull(f.get());
+    }
+
+    @Test
+    public void testRetrieve_WhenSubmitterMemberDown() throws ExecutionException, InterruptedException {
+        String name = randomString();
+
+        DurableExecutorService executorService = client.getDurableExecutorService(name);
+        SleepingTask task = new SleepingTask(4);
+        long taskId = executorService.submit(task).getTaskId();
+
+        client.shutdown();
+        client = hazelcastFactory.newHazelcastClient();
+
+        executorService = client.getDurableExecutorService(name);
+        Future<Boolean> future = executorService.retrieveResult(taskId);
+        assertTrue(future.get());
+    }
+
+    @Test
+    public void testRetrieve_WhenOwnerMemberDown() throws ExecutionException, InterruptedException {
+        String name = randomString();
+        String key = generateKeyOwnedBy(instance2);
+
+        DurableExecutorService executorService = client.getDurableExecutorService(name);
+        SleepingTask task = new SleepingTask(4);
+        long taskId = executorService.submitToKeyOwner(task, key).getTaskId();
+        instance2.shutdown();
+
+        Future<Boolean> future = executorService.retrieveResult(taskId);
+        assertTrue(future.get());
+    }
+
+    @Test
+    public void testRetrieve_WhenResultOverwritten() throws ExecutionException, InterruptedException {
+        String name = randomString();
+        DurableExecutorService executorService = client.getDurableExecutorService(name);
+        DurableExecutorServiceFuture<String> future = executorService.submitToKeyOwner(new BasicTestCallable(), name);
+        long taskId = future.getTaskId();
+        future.get();
+
+        for (int i = 0; i < DEFAULT_RING_BUFFER_CAPACITY; i++) {
+            executorService.submitToKeyOwner(new BasicTestCallable(), name);
+        }
+
+        Future<Object> f = executorService.retrieveResult(taskId);
+        try {
```

Comment:
it will be a `StaleSequenceException` wrapped in `ExecutionException`, that's why I'm asserting the cause of the exception, not the original exception